src/test/java/uk/gov/justice/digital/zone/StructuredZoneTest.java

```diff
@@ -0,0 +1,189 @@
+    @BeforeEach
+    void setUp() {
+        sourceRefStatic = mockStatic(SourceReferenceService.class);
+    }
+
+    @AfterEach
+    void tearDown() {
+        sourceRefStatic.close();
+    }
+
+    @Test
+    public void shouldReturnValidStructuredS3Path() {
+        val source = "oms_owner";
+        val table  = "agency_internal_locations";
+        val operation = "load";
+        val expectedStructuredS3Path = String.join("/", S3_PATH, source, table, operation);
+        assertEquals(expectedStructuredS3Path, this.storage.getTablePath(S3_PATH, source, table, operation));
+    }
+
+    @Test
+    @ExtendWith(MockitoExtension.class)
+    public void shouldThrowExceptionWhenViolationsPathNotSet() {
+        DataStorageService storage1 = mock(DataStorageService.class);
+        assertThrows(IllegalStateException.class, () ->
+                new StructuredZone(jobParameters, storage1), "Expected to throw exception, but it didn't"
+        );
+    }
+
+    @Test
+    @ExtendWith(MockitoExtension.class)
+    public void shouldHandleValidRecords() {
+        // Define a schema for the row
+        StructType schema = new StructType()
+                .add("source", StringType, false)
+                .add("table", StringType, false)
+                .add("operation", StringType, false);
+        // Create a Row object with key-value pairs
+        Row table = new GenericRowWithSchema(Arrays.asList("oms_owner", "agency_internal_locations", "load").toArray(),
+                schema);
+        SourceReference ref = mock(SourceReference.class);
+        JobParameters jobParams = mock(JobParameters.class);
+        DataStorageService storage1 = mock(DataStorageService.class);
+        doReturn("testPath").when(storage1).getTablePath(S3_PATH, ref);
+        sourceRefStatic.when(() -> SourceReferenceService
+                        .getSourceReference(table.getAs("source"), table.getAs("table")))
+                .thenReturn(optionalRef);
+        doReturn(true).when(optionalRef).isPresent();
+        doReturn(ref).when(optionalRef).get();
+
+        doReturn(Optional.of("s3://loadjob/violations")).when(jobParams).getViolationsS3Path();
+        doReturn(Optional.of("s3://loadjob/structured")).when(jobParams).getStructuredS3Path();
+        StructuredZone structuredZoneTest = spy(new StructuredZone(jobParams, storage1));
+        doReturn(mockedDataSet).when(structuredZoneTest).validateJsonData(any(), any(), any(), any());
+        doNothing().when(structuredZoneTest).handleInValidRecords(any(), any(), any(), any(), any());
+        doReturn(mockedDataSet).when(structuredZoneTest).handleValidRecords(any(), any(), any());
+        when(mockedDataSet.count()).thenReturn(10L);
+        Dataset<Row> actual_result = structuredZoneTest.process(spark, mockedDataSet, table);
+        assertNotNull(actual_result);
+    }
+
+    @Test
+    @ExtendWith(MockitoExtension.class)
+    public void shouldHandleInValidRecords() {
+        // Define a schema for the row
```

Comment:
These comments are effectively duplicating the code below. When writing comments you typically only need them where the `why` needs to be made clear. The `what` should be self evident from the code itself.